pkg/networkmanager/interfaces.js

```diff
@@ -207,6 +207,10 @@ function NetworkManagerModel() {
 
     var self = this;
 
+    /* resolved once first stage of initialization is done */
+    self._preinit_resolve = null;
+    self.preinit = new Promise((resolve, reject) => { self._preinit_resolve = resolve });
```

Comment:
What about moving the `client.call` from below to here?  That way we wouldn't have to introduce the `self._preinit_resolve` variable.

Like so:

```
self.preinit = new Promise((resolve, reject) => {
   client.call("/org/freedesktop/NetworkManager",
                "org.freedesktop.DBus.Properties", "Get",
                ["org.freedesktop.NetworkManager", "State"], { flags: "" })
            .fail(complain)
            .done(function(reply, options) {
                if (options.flags) {
                    if (options.flags.indexOf(">") !== -1)
                        utils.set_byteorder("be");
                    else if (options.flags.indexOf("<") !== -1)
                        utils.set_byteorder("le");
                    resolve();
                }
            });
});
```

Reply:
You are right, this is much more elegant. Fixed!